src/Fantomas.Core/CodePrinter.fs

```diff
@@ -3262,72 +3264,72 @@ let genTypeDefn (td: TypeDefn) =
-            +> col sepSemi node.Fields genField
-            +> addSpaceIfSpaceAroundDelimiter
-            +> genSingleTextNode node.ClosingBrace
-
         let multilineExpression (ctx: Context) =
+            let genRecordFields =
+                genSingleTextNode node.OpeningBrace
+                +> indentSepNlnUnindent (atCurrentColumn (col sepNln node.Fields genField))
+                +> sepNln
+                +> genSingleTextNode node.ClosingBrace
+
+            let genMembers =
+                onlyIf hasMembers sepNln
+                +> sepNlnTypeAndMembers typeDefnNode
+                +> genMemberDefnList members
+
             let aligned =
-                let msIsEmpty = List.isEmpty members
-
-                (ifElseCtx
-                    (fun ctx -> ctx.Config.ExperimentalStroustrupStyle && msIsEmpty)
-                    (genAccessOpt node.Accessibility)
-                    (opt (indent +> sepNln) node.Accessibility genSingleTextNode)
-                 +> genSingleTextNode node.OpeningBrace
-                 +> indentSepNlnUnindent (atCurrentColumn (col sepNln node.Fields genField))
-                 +> sepNln
-                 +> genSingleTextNode node.ClosingBrace
-                 +> onlyIfCtx
-                     (fun ctx -> not (ctx.Config.ExperimentalStroustrupStyle && msIsEmpty))
-                     (optSingle (fun _ -> unindent) node.Accessibility)
-                 +> onlyIf (List.isNotEmpty members) sepNln
-                 +> sepNlnTypeAndMembers typeDefnNode
-                 +> genMemberDefnList members)
-                    ctx
+                opt (indent +> sepNln) node.Accessibility genSingleTextNode
+                +> genRecordFields
+                +> optSingle (fun _ -> unindent) node.Accessibility
+                +> genMembers
 
             let anyFieldHasXmlDoc =
-                List.exists (fun (fieldNode: FieldNode) -> fieldNode.XmlDoc.IsSome) node.Fields
-
-            match ctx.Config.MultilineBracketStyle with
-            | Aligned
-            | ExperimentalStroustrup -> aligned
-            | Cramped when anyFieldHasXmlDoc -> aligned
-            | Cramped ->
-                (sepNlnUnlessLastEventIsNewline
-                 +> opt (indent +> sepNln) node.Accessibility genSingleTextNode
-                 +> genSingleTextNodeSuffixDelimiter node.OpeningBrace
-                 +> atCurrentColumn (sepNlnWhenWriteBeforeNewlineNotEmpty +> col sepNln node.Fields genField)
-                 +> addSpaceIfSpaceAroundDelimiter
-                 +> genSingleTextNode node.ClosingBrace
-                 +> optSingle (fun _ -> unindent) node.Accessibility
-                 +> onlyIf (List.isNotEmpty members) sepNln
-                 +> sepNlnTypeAndMembers typeDefnNode
-                 +> genMemberDefnList members)
-                    ctx
+                node.Fields |> List.exists (fun fieldNode -> fieldNode.XmlDoc.IsSome)
+
+            let genMultilineExpression =
```

Comment:
Don't introduce a function value only to call it immediately as the next statement.

Refactor to
```fsharp
            match ctx.Config.MultilineBracketStyle with
            | ExperimentalStroustrup when hasNoMembers ->
                (genAccessOpt node.Accessibility +> genRecordFields +> genMembers) ctx
            | Aligned
            | ExperimentalStroustrup -> aligned ctx
            | Cramped when anyFieldHasXmlDoc -> aligned ctx
            | Cramped ->
                (sepNlnUnlessLastEventIsNewline
                 +> opt (indent +> sepNln) node.Accessibility genSingleTextNode
                 +> genSingleTextNodeSuffixDelimiter node.OpeningBrace
                 +> atCurrentColumn (sepNlnWhenWriteBeforeNewlineNotEmpty +> col sepNln node.Fields genField)
                 +> addSpaceIfSpaceAroundDelimiter
                 +> genSingleTextNode node.ClosingBrace
                 +> optSingle (fun _ -> unindent) node.Accessibility
                 +> onlyIf hasMembers sepNln
                 +> sepNlnTypeAndMembers typeDefnNode
                 +> genMemberDefnList members)
                    ctx
```
instead.

Reply:
You'd prefer that even when it makes us have to wrap the more complicated expressions here in parentheses? IMHO it's more clear and readable without the additional parentheses when we can take advantage of partial application like that but I'm totally fine to change it back to that if you'd prefer.

Reply:
You currently have something like:
```fsharp
let multilineExpression (ctx: Context) =
    let genMultilineExpression = !- "code ..."
    genMultilineExpression ctx
```
which is the equivalent of storing the return value only to return it in the next line. That bothers me.

As a middle ground, I propose to introduce some more intermediate values.
Extract the clauses bodies with parentheses to `stroustrupWithoutMembers` and `cramped`.

End the function with:
```fsharp
match ctx.Config.MultilineBracketStyle with
| ExperimentalStroustrup when hasNoMembers -> stroustrupWithoutMembers ctx
| Aligned
| ExperimentalStroustrup -> aligned ctx
| Cramped when anyFieldHasXmlDoc -> aligned ctx
| Cramped -> cramped ctx
```